components/kyma-environment-broker/internal/process/provisioning/input/input.go

```diff
@@ -30,13 +32,39 @@ func (r *RuntimeInput) SetProvisioningParameters(params internal.ProvisioningPar
 	return r
 }
 
+// SetOverrides sets the overrides for the given component and discard the previous ones.
+//
+// Deprecated: use AppendOverrides
 func (r *RuntimeInput) SetOverrides(component string, overrides []*gqlschema.ConfigEntryInput) internal.ProvisionInputCreator {
-	// TODO add possibility to adding same overrides to the same component
+	r.mutex.Lock("SetOverrides")
+	defer r.mutex.Unlock("SetOverrides")
+
 	r.overrides[component] = overrides
 	return r
 }
 
+// AppendOverrides appends overrides for the given components, the existing overrides are preserved.
+func (r *RuntimeInput) AppendOverrides(component string, overrides []*gqlschema.ConfigEntryInput) internal.ProvisionInputCreator {
+	r.mutex.Lock("AppendOverrides")
```

Comment:
SetOverrides and AppendOverrides works with overrides field but uses different mutex name which works like different mutex. In my opinion it should be the same name.